File: keyple-plugin-android-nfc/src/main/java/org/eclipse/keyple/plugin/androidnfc/AndroidNfcPlugin.java
/*
 * Copyright (c) 2018 Calypso Networks Association https://www.calypsonet-asso.org/
 *
 * All rights reserved. This program and the accompanying materials are made available under the
 * terms of the Eclipse Public License version 2.0 which accompanies this distribution, and is
 * available at https://www.eclipse.org/org/documents/epl-2.0/EPL-2.0.html
 */

package org.eclipse.keyple.plugin.androidnfc;

import java.util.ArrayList;
import java.util.List;
import org.eclipse.keyple.seproxy.ProxyReader;
import org.eclipse.keyple.seproxy.ReadersPlugin;
import android.util.Log;


/**
 * Readers Plugin for Android platform based on the NFC Adapter
 */
public class AndroidNfcPlugin implements ReadersPlugin {

    private static final String TAG = AndroidNfcPlugin.class.getSimpleName();

    private final static AndroidNfcPlugin uniqueInstance = new AndroidNfcPlugin();

    private ProxyReader reader;

    private AndroidNfcPlugin() {

        if (this.reader == null) {
            Log.i(TAG, "Instanciate singleton NFC Plugin");
            this.reader = AndroidNfcReader.getInstance();
        }

    }

    public static AndroidNfcPlugin getInstance() {
        return uniqueInstance;
    }


    @Override
    public String getName() {
        return "AndroidNFCPlugin";
    }

    /**
     * For an Android NFC device, the Android NFC Plugin manages only one @{@link AndroidNfcReader}.
     * 
     * @return List<ProxyReader> : contains only one element, the
     *         singleton @{@link AndroidNfcReader}
     */
    @Override
    public List<ProxyReader> getReaders() {
        // return the only one reader in a list
        List<ProxyReader> readers = new ArrayList<ProxyReader>();
        readers.add(reader);
        return readers;
    }
}

File: android/keyple-plugin/android-nfc/src/main/kotlin/org/eclipse/keyple/plugin/android/nfc/AndroidNfcPlugin.kt
/********************************************************************************
 * Copyright (c) 2020 Calypso Networks Association https://www.calypsonet-asso.org/
 *
 * See the NOTICE file(s) distributed with this work for additional information regarding copyright
 * ownership.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse
 * Public License 2.0 which is available at http://www.eclipse.org/legal/epl-2.0
 *
 * SPDX-License-Identifier: EPL-2.0
 ********************************************************************************/
package org.eclipse.keyple.plugin.android.nfc

import org.eclipse.keyple.core.service.Plugin

/**
 * Provides the public elements used to manage the Android NFC plugin.<br>
 * It defines in particular the type of object produced by the {@link AndroidNfcPluginFactory} and allow
 * to set parameters to retrieve the communication mode from the name of a reader.
 */
interface AndroidNfcPlugin : Plugin {
    companion object {
        const val PLUGIN_NAME = "AndroidNfcPlugin"
    }
}
